src/hooks/core/chat/useDMMessage.ts

```diff
@@ -120,7 +123,7 @@ const useDMMessage = () => {
       await Promise.all(
         (channel?.messages || []).map(async (message) => {
           if (message?.type === 'deleted') return;
-          const chat = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
+          const chat = ChatSchema.fromGetAllChannelAPI(channel?.id, message?.message);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces optional chaining on `message?.message` when calling `ChatSchema.fromGetAllChannelAPI(channel?.id, message?.message)`. This could potentially pass `undefined` as an argument if `message` is undefined. Ensure that your function can handle this case correctly, or add a check to prevent this scenario.

```diff
- const chat = ChatSchema.fromGetAllChannelAPI(channel?.id, message?.message);
+ if (message?.message) {
+   const chat = ChatSchema.fromGetAllChannelAPI(channel?.id, message.message);
+   await chat.save(localDb);
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->